sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -660,13 +895,39 @@ public CarbonWriter build() throws IOException, InvalidLoadOptionException {
       // removed from the load. LoadWithoutConverter flag is going to point to the Loader Builder
       // which will skip Conversion Step.
       loadModel.setLoadWithoutConverterStep(true);
-      return new AvroCarbonWriter(loadModel, hadoopConf, this.avroSchema);
+      AvroCarbonWriter avroCarbonWriter = new AvroCarbonWriter(loadModel,
```

Comment:
We have some code duplications for each type of writer. Suggest to refactor it. Something like this - 
```suggestion
    CarbonWriter carbonWriter;
    if (this.writerType == WRITER_TYPE.AVRO) {
      // AVRO records are pushed to Carbon as Object not as Strings. This was done in order to
      // handle multi level complex type support. As there are no conversion converter step is
      // removed from the load. LoadWithoutConverter flag is going to point to the Loader Builder
      // which will skip Conversion Step.
      loadModel.setLoadWithoutConverterStep(true);
      carbonWriter = new AvroCarbonWriter(loadModel, hadoopConf, this.avroSchema);
    } else if (this.writerType == WRITER_TYPE.JSON) {
      loadModel.setJsonFileLoad(true);
      carbonWriter = new JsonCarbonWriter(loadModel, hadoopConf);
    } else if (this.writerType == WRITER_TYPE.PARQUET) {
      loadModel.setLoadWithoutConverterStep(true);
      carbonWriter = new ParquetCarbonWriter(loadModel, hadoopConf, this.avroSchema);
    } else if (this.writerType == WRITER_TYPE.ORC) {
      carbonWriter = new ORCCarbonWriter(loadModel, hadoopConf);
    } else {
      // CSV
      CSVCarbonWriter csvCarbonWriter = new CSVCarbonWriter(loadModel, hadoopConf);
      if (!this.options.containsKey(CarbonCommonConstants.FILE_HEADER)) {
        csvCarbonWriter.setSkipHeader(true);
      }
      carbonWriter = csvCarbonWriter;
    }
    if (!StringUtils.isEmpty(filePath)) {
      carbonWriter.validateAndSetDataFiles(this.dataFiles);
    }
    return carbonWriter;
```

Reply:
done